File: design-pattern-jvm/app/src/main/java/design/pattern/java/patterns/creational/abstract_factory_method/ToyotaFactory.java
package design.pattern.java.patterns.creational.abstract_factory_method;

public class ToyotaFactory implements VehicleFactory{

  @Override
  public DeportVehicle createDeportCar() {
    return new ToyotaYaris();
  }

  @Override
  public RusticVehicle createRusticCar() {
    return new ToyotaForRunner();
  }
  
}

File: design-pattern-jvm/app/src/main/kotlin/design/pattern/kotlin/patterns/creational/abstract_factory_method/ToyotaFactory.kt
package design.pattern.kotlin.patterns.creational.abstract_factory_method

class ToyotaFactory : VehicleFactory {
    override fun createDeportCar(): DeportVehicle {
        return ToyotaYaris()
    }

    override fun createRusticCar(): RusticVehicle {
        return ToyotaForRunner()
    }
}